assert/utils_test.go

```diff
@@ -2549,6 +2549,140 @@ func TestFormatContainSubstringError(t *testing.T) {
 	})
 }
 
+func TestFindExactCaseMismatch(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name     string
+		text     string
+		substr   string
+		exists   bool
+		position int
+		found    string
+	}{
+		{
+			name:     "Found exact case mismatch",
+			text:     "Hello, WORLD!",
+			substr:   "world",
+			exists:   true,
+			position: 7,
+			found:    "WORLD",
+		},
+		{
+			name:     "Found case mismatch at beginning",
+			text:     "HELLO world",
+			substr:   "hello",
+			exists:   true,
+			position: 0,
+			found:    "HELLO",
+		},
+		{
+			name:     "Found case mismatch with complex text",
+			text:     "Get \"http://127.0.0.1:56748\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)",
```

Comment:
Use string literals here, it would be clearer


```suggestion
			text:     `Get "http://127.0.0.1:56748": context deadline exceeded (Client.Timeout exceeded while awaiting headers)`,
```